### PR TITLE
[Dist] enable to specify sort_etype for sample_etype_neighbours

### PR DESCRIPTION
**Pull request by Rhett-Ying**
## Description

follow-up for https://github.com/dmlc/dgl/issues/4150.
## Checklist
Please feel free to remove inapplicable items for your PR.
- [ ] The PR title starts with [$CATEGORY] (such as [NN], [Model], [Doc], [Feature]])
- [ ] Changes are complete (i.e. I finished coding on this PR)
- [ ] All changes have test coverage
- [ ] Code is well-documented
- [ ] To the best of my knowledge, examples are either not affected by this change,
      or have been fixed to be compatible with this change
- [ ] Related issue is referred in this PR
- [ ] If the PR is for a new model/paper, I've updated the example index [here](../examples/README.md).

### CODE REVIEW

File: python/dgl/distributed/graph_services.py [hunk 1]
```diff
@@ -80,7 +80,9 @@ def _sample_etype_neighbors(local_g, partition_book, seed_nodes, etype_field,
     """
     local_ids = partition_book.nid2localnid(seed_nodes, partition_book.partid)
     local_ids = F.astype(local_ids, local_g.idtype)
-
+    sort_etypes = os.environ.get('DGL_SORT_ETYPES', '')
```
Comment (jermainewang) [inline python/dgl/distributed/graph_services.py:83]: This is really bad practice. Please do not rely on environment variable besides the very entry point. This is very confusing and difficult for others to understand when and where this flag is set. Please pass it down by function argument.

**Review by jermainewang — CHANGES_REQUESTED** [review]

File: python/dgl/distributed/dist_context.py [hunk 2]
```diff
@@ -227,6 +227,7 @@ def initialize(ip_config, num_servers=1, num_workers=0,
             'Please define DGL_CONF_PATH to run DistGraph server'
         formats = os.environ.get('DGL_GRAPH_FORMAT', 'csc').split(',')
         formats = [f.strip() for f in formats]
+        sort_etypes = os.environ.get('DGL_SORT_ETYPES', '')
```
[the remark this answers is not in the document]
Comment (Rhett-Ying) [reply python/dgl/distributed/dist_context.py:230]: Do we have corresponding doc? where is it?

**Comment by zheng-da** [comment]
the main question is why we should expose the sort_etypes flag to users? if we provide this interface, users need to specify this flag and the graph format correctly. this will be very confusing to users.
we should enable sort_etypes by default based on the graph formats specified by users if the input graph is a heterogeneous graph.

File: python/dgl/distributed/dist_graph.py [hunk 3]
```diff
@@ -339,6 +344,12 @@ def __init__(self, server_id, ip_config, num_servers,
             # Create the graph formats specified the users.
             self.client_g = self.client_g.formats(graph_format)
             self.client_g.create_formats_()
+            if sort_etypes == 'csr':
+                self.client_g = sort_csr_by_tag(
+                    self.client_g, tag=self.client_g.edata[ETYPE], tag_type='edge')
```
Comment (Rhett-Ying) [inline python/dgl/distributed/dist_graph.py:349]: @jermainewang @zheng-da Is sorting like this correct ? Is passing tag with `g.edata[ETYPE]` correct? Do I need to sort the inner edges only?

**Review by jermainewang — APPROVED** [review]

**Pull request merged**